scripts/release-merge-to-main.ts

```diff
@@ -0,0 +1,178 @@
+// This file creates PRs when releases are published to merge back to the main
+// branch.
+
+// Environment:
+//   GITHUB_REPOSITORY, GITHUB_EVENT_PATH, and others
+//     See https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
+//   GITHUB_TOKEN: GitHub authorization token.
+//     Must have write permissions for `actions`, `contents`, `pull_requests`.
+
+import fs from 'fs';
+
+import { RequestError } from 'octokit';
+
+import { getOctokit } from './lib/dependencies';
+
+/**
+ * Valid value for an environment variable.
+ */
+type EnvironmentVariableName =
+  'GITHUB_REPOSITORY' |
+  'GITHUB_EVENT_PATH' |
+  'GITHUB_TOKEN';
+
+/**
+ * Partial contents of the event payload, for a release event.
+ */
+interface GitHubReleasePayload {
+  release: {
+    tag_name: string;
+  }
+}
+
+/**
+ * The name of the branch to merge into.
+ */
+const base = 'main';
+
+/**
+ * Read the environment variable, or throw an error.
+ * @param variable The environment variable to look up.
+ * @returns The environment variable value.
+ */
+function getEnv(variable: EnvironmentVariableName): string {
+  const result = process.env[variable];
+
+  if (typeof result !== 'string') {
+    throw new ReferenceError(`Environment variable ${ variable } is not set`);
+  }
+
+  return result;
+}
+
+/**
+ * Ensure that the given branch exists, and points to the given tag.
+ * @param owner The repository owner.
+ * @param repo The repository name (without the owner).
+ * @param branchName The name of the branch.
+ * @param tagName The name of the tag.
+ */
+async function ensureBranch(owner: string, repo: string, branchName: string, tagName: string): Promise<void> {
+  const ref = `heads/${ branchName }`;
+  const { git } = getOctokit().rest;
+  const { data: tagRef } = await git.getRef({
+    owner, repo, ref: `tags/${ tagName }`,
+  });
+  const { sha } = tagRef.object;
+
+  try {
+    const { data: existingBranch } = await git.getRef({
+      owner, repo, ref,
+    });
+
+    if (existingBranch.object.sha !== sha) {
+    // Branch exists, but points at the wrong hash; update it.
```

Comment:
```suggestion
      // Branch exists, but points at the wrong hash; update it.
```

Maybe also add a log message here with old and new hash?